### PR TITLE
feat: davinci with https support

### PR DESCRIPTION
**Pull request by jferrl**
This PR includes native support to create a https server with davinci

### CODE REVIEW

**Review by joseandrespg — APPROVED** [review]

File: packages/core/src/createApp.ts [hunk 1]
```diff
@@ -23,6 +24,11 @@ interface HealthChecksOptions {
 	readynessEndpoint?: string;
 }
 
+interface DaVinciTlsOptions {
```
Comment (pierissimo) [inline packages/core/src/createApp.ts:27]: we can put key and cert inline, no need to create a separate interface

File: packages/core/src/createApp.ts [hunk 2]
```diff
@@ -39,6 +45,7 @@ export interface DaVinciOptions {
 			options?: any;
 		};
 	};
+	tls?: DaVinciTlsOptions;
```
Comment (pierissimo) [inline packages/core/src/createApp.ts:48]: can we rename the option to `https`?

Additionally, you can probably use the types from https.ServerOptions
```ts
https?: https.ServerOptions;
```

File: packages/core/src/createApp.ts [hunk 3]
```diff
@@ -93,7 +100,7 @@ export const configureExpress = async (app, options: DaVinciOptions = {}, runMid
 			// eslint-disable-next-line
 			const swaggerUi = require('swagger-ui-express');
 			app.use(swaggerUIPath, swaggerUi.serve, swaggerUi.setup(fullSwaggerDoc, swaggerUIOpts));
-			console.log(`--- Swagger UI available at ${swaggerUIPath}`);
+			debug(`--- Swagger UI available at ${swaggerUIPath}`);
```
Comment (pierissimo) [inline packages/core/src/createApp.ts:103]: let's revert this

File: packages/core/src/createApp.ts [hunk 4]
```diff
@@ -130,6 +137,16 @@ export const configureTerminus = (app, healthChecks: HealthChecksOptions = {}) =
 	return createTerminus(app.server, terminusOptions);
 };
 
+const existsTlsOptions = (options: DaVinciOptions) => options.tls && options.tls.key && options.tls.cert;
```
Comment (pierissimo) [inline packages/core/src/createApp.ts:140]: better named `hasHttpsOptions`

File: packages/core/src/createApp.ts [hunk 5]
```diff
@@ -153,7 +170,7 @@ export const createApp = (...args: CreateAppArgs): Promise<DaVinciExpress> => {
 
 		await new Promise<void>(resolve =>
 			server.listen(config.PORT, () => {
-				console.log(`--- Server listening on ${config.PORT}`);
+				debug(`--- Server listening on ${config.PORT}`);
```
Comment (pierissimo) [inline packages/core/src/createApp.ts:173]: same as above, let's revert this

**Review by pierissimo — CHANGES_REQUESTED** [review]

File: packages/core/src/createApp.ts (hunk 2 above)
Comment (jferrl) [reply packages/core/src/createApp.ts]: Done!

File: packages/core/src/createApp.ts (hunk 3 above)
Comment (jferrl) [reply packages/core/src/createApp.ts]: Done!

File: packages/core/src/createApp.ts (hunk 5 above)
Comment (jferrl) [reply packages/core/src/createApp.ts]: Done!

File: packages/core/src/createApp.ts (hunk 4 above)
Comment (jferrl) [reply packages/core/src/createApp.ts]: Done!

**Review by pierissimo — APPROVED** [review]

**Pull request merged**